temporary-table.md

```diff
@@ -0,0 +1,332 @@
+
+After `users` is created in session A, session B can also read from and write to the `users` table:
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT * FROM users;
+```
+
+```
+Empty set (0.00 sec)
+```
+
+> **Note:**
+>
+> If the transaction is automatically committed, after the SQL statement is executed, the inserted data is automatically cleared and unavailable to subsequent SQL executions. Therefore, you should use non-autocommit transactions to read from and write to global temporary tables.
+
+## Limit the memory usage of temporary tables
+
+No matter which storage engine is declared as `ENGINE` when you define a table, the data of local temporary tables and global temporary tables is only stored in the memory of TiDB instances. This data is not persisted.
+
+To avoid memory overflow, you can limit the size of each temporary table using the [`tidb_tmp_table_max_size`](/system-variables.md#tidb_tmp_table_max_size-new-in-v530) system variable. Once a temporary table is larger than the `tidb_tmp_table_max_size` threshold value, TiDB reports an error. The default value of `tidb_tmp_table_max_size` is `64MB`.
+
+For example, set the maximum size of a temporary table to `256MB`:
+
+{{< copyable "sql" >}}
+
+```sql
+SET GLOBAL tidb_tmp_table_max_size=268435456;
+```
+
+## Compatibility restrictions with other TiDB features
+
+Local temporary tables and global temporary tables in TiDB are NOT compatible with the following TiDB features:
+
+- `AUTO_RANDOM` columns
+- `SHARD_ROW_ID_BITS` and `PRE_SPLIT_REGIONS` table options
+- Partitioned tables
+- `SPLIT REGION` statements
+- `ADMIN CHECK TABLE` and `ADMIN CHECKSUM TABLE` statements
+- `FLASHBACK TABLE` and `RECOVER TABLE` statements
+- Executing `CREATE TABLE LIKE` statements based on a temporary table
+- Stale Read
+- Foreign keys
+- SQL bindings
+- TiFlash replicas
+- Creating views on a temporary table
+- Placement Rules
+- Execution plans involving a temporary table are not cached by `prepare plan cache`.
+
+Local temporary tables in TiDB do not support the following feature:
+
+- Reading historical data using the `tidb_snapshot` system variable.
+
+## TiDB ecosystem tool support
+
+Local temporary tables are not exported, backed up or replicated by TiDB ecosystem tools, because these tables are visible only to the current session.
+
+Global temporary tables are exported, backed up, and replicated by TiDB ecosystem tools, because the table definition is globally visible. Note that the data on the tables are not exported.
+
+> **Note:**
+>
+> - Replicating temporary tables using TiCDC requires TiCDC v5.3.0 or later. Otherwise, the table definition of the downstream table is wrong.
+> - Backing up temporary tables using BR requires BR v5.3.0 or later. Otherwise, the table definitions of the backed temporary tables are wrong.
+> - The cluster to export, the cluster after data restore, and the downstream cluster of a replication should support global temporary tables. Otherwise, an error is reported.
```

Comment:
```suggestion
> - Replicating temporary tables using TiCDC requires TiCDC v5.3.0 or a later version. Otherwise, the table definition of the downstream table is wrong.
> - Backing up temporary tables using BR requires BR v5.3.0 or a later version. Otherwise, the table definitions of the backed temporary tables are wrong.
> - The cluster to export, the cluster after data restore, and the downstream cluster of a replication should support global temporary tables. Otherwise, an error is reported.
```